crates/net/discv4/src/lib.rs

```diff
@@ -695,6 +714,20 @@ impl Discv4Service {
         }
     }
 
+    /// Handler for incoming `EnrResponse` message
+    fn on_enr_response(&mut self, msg: EnrResponse, remote_addr: SocketAddr, node_id: PeerId) {
+        match self.pending_enr_requests.entry(node_id) {
+            Entry::Occupied(entry) => entry.remove(),
+            Entry::Vacant(_) => return,
+        };
```

Comment:
Should we just extend the `NodeEntry` with the contents map and set this on_enr_response?

or better convert the contents map into something that's more explicit like has `eth` or something?